prime-router/metadata/hl7_mapping/common/observation-result.yml

```
@@ -165,30 +165,30 @@ elements:
     hl7Spec: [ '%{hl7OBXField}-19' ]
 
   - name: performing-organization-name-pracrole
-    condition: '%observation.performer.resolve() is PractitionerRole'
-    resource: '%resource.performer.resolve().organization.resolve()'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists() and %observation.performer.resolve().ofType(PractitionerRole).organization.exists()'
```

@jimmyfagan Can you double check me on this?  The logic is not that we will attempt to populate OBX-23 using the PractionerRole from the performer array if there is a PractionerRole and it has an organization otherwise we'll use the Organization

---

These look okay, and clearly work based on the Translation Tests. They are a bit hard to understand without looking at the mappings to be honest though. Here and in `performing-organization-address-pracrole`, I might recommend something like the following, though optionally since in practice they should produce the same results:
```suggestion
    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists() and %observation.performer.resolve().ofType(PractitionerRole).code.coding.code = "MDIR"'
```